bundles/net.tourbook/src/net/tourbook/importdata/IRawDataReader.java

```diff
@@ -1,66 +1,68 @@
+   /**
+    * @return Returns the complete data size which will be received on the COM port for this device
+    */
+   public int getTransferDataSize();
 
-	/**
-	 * Read the data from the raw data file and create the device and tour data
-	 * 
-	 * @param importFilePath
-	 * @param deviceData
-	 * @param alreadyImportedTours
-	 *            Contains all tours which are already imported and displayed in the import view.
-	 *            Tour id is the hash map key. Newly imported tour should not be added to this map,
-	 *            they are added after this method is returned, they must be put into the parameter
-	 *            <i>importedTours</i> map.
-	 *            <p>
-	 *            This map can be used to check if a tour is already imported and displayed in the
-	 *            import view.
-	 * @param newlyImportedTours
-	 *            Contains all tours which are imported by this method.
-	 * @return Returns <code>true</code> when the import was successfull, the parameters
-	 *         <code>deviceData</code> and <code>tourData</code> are set from the imported file.
-	 */
-	public boolean processDeviceData(	String importFilePath,
-										DeviceData deviceData,
-										HashMap<Long, TourData> alreadyImportedTours,
-										HashMap<Long, TourData> newlyImportedTours);
-
-	/**
-	 * Validate data format
-	 * 
-	 * @param importFilePath
-	 *            file name for the file which is validated
-	 * @return return <code>true</code> when the file has the format for this device reader
-	 */
-	public boolean validateRawData(String importFilePath);
+   /**
+    * Read the data from the raw data file and create the device and tour data
+    *
+    * @param importFilePath
+    * @param deviceData
+    * @param alreadyImportedTours
+    *           Contains all tours which are already imported and displayed in the import view.
+    *           Tour id is the hash map key. Newly imported tour should not be added to this map,
+    *           they are added after this method is returned, they must be put into the parameter
+    *           <i>importedTours</i> map.
+    *           <p>
+    *           This map can be used to check if a tour is already imported and displayed in the
+    *           import view.
+    * @param newlyImportedTours
+    *           Contains all tours which are imported by this method.
+    * @param isReimport
+    *           True if this tour is re-imported, false otherwise
+    * @return Returns <code>true</code> when the import was successful, the parameters
+    *         <code>deviceData</code> and <code>tourData</code> are set from the imported file.
+    */
+   public boolean processDeviceData(String importFilePath,
+                                    DeviceData deviceData,
+                                    Map<Long, TourData> alreadyImportedTours,
+                                    Map<Long, TourData> newlyImportedTours,
+                                    final boolean isReimport);

```

Comment:
@wolfgang-ch Because I added this parameter to know if the file import is a new import or a re-import, that cascaded all the other changes in the *DataReaders